hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java

```diff
@@ -375,21 +402,50 @@ private void consume(final RecordStreamObject object) {
 						.build()
 		);
 
-		// In phase 2 we will need to add logic here to also
-		// save the sidecar records of the current object
+		final var sidecars = object.getSidecars();
+		if (isNotEmpty(sidecars)) {
+			for (final var sidecar : sidecars) {
+				if (sidecar.getSidecarRecordsCase() == SidecarRecordsCase.STATE_CHANGES
+						|| sidecar.getSidecarRecordsCase() == SidecarRecordsCase.ACTIONS
+						|| sidecar.getSidecarRecordsCase() == SidecarRecordsCase.BYTECODE) {
```

Comment:
Agreed, but since we have to keep track of which sidecar types are present in the current sidecar file, I don't think there is a way to do it without having specific checks to convert from `TransactionSidecarRecord.sidecarRecordCase` -> `SidecarType` ?